src/shogun/distance/EuclideanDistance.cpp

```diff
@@ -37,80 +36,76 @@ CEuclideanDistance::~CEuclideanDistance()
 
 bool CEuclideanDistance::init(CFeatures* l, CFeatures* r)
 {
-	CRealDistance::init(l, r);
+	cleanup();
+
+	CDistance::init(l, r);
+	REQUIRE(l->has_property(FP_DOT), "Left hand side features must support dot property!\n");
+	REQUIRE(r->has_property(FP_DOT), "Right hand side features must support dot property!\n");
+
+	CDotFeatures* casted_l=static_cast<CDotFeatures*>(l);
+	CDotFeatures* casted_r=static_cast<CDotFeatures*>(r);
+
+	REQUIRE(casted_l->get_dim_feature_space()==casted_r->get_dim_feature_space(),
+		"Number of dimension mismatch (l:%d vs. r:%d) in feature space!\n",
```

Comment:
Maybe dont say feature space here, as that has a meaning in kernel land, and it is not the dimension of the input data.

"Dimension mismatch (left, right)" should do it